### PR TITLE
added new bdp spark test

### PR DESCRIPTION
**Pull request by korry8911**
bdp spark smoke test.  Creates 3 node bdp cluster and 2 node spark cluster within bdp cluster, then submits pi.py example to ensure proper communication of spark master and worker.

### CODE REVIEW

File: tests/bdp_spark_test.erl [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+confirm() ->
+
+    %build cluster
+    lager:info("Building cluster"),
+    ClusterSize = 3,
+    ClusterConfig = [{lager, [{handlers, [{file, "console.log"}, {level, debug}] }]}],
+    Nodes = [Node1, Node2, _Node3] = rt:deploy_nodes(ClusterSize, ClusterConfig),
+    rt:join_cluster(Nodes),
+    %let cluster settle down
+    lager:info("Waiting 20 seconds..."),
```
Comment (hmmr) [inline tests/bdp_spark_test.erl]: No, a simple `timer:sleep/1` will never guarantee that the condition you are waiting for, will be met.

Have a look at various rt:wait_for_* functions, as used (for example) here: https://github.com/basho/riak_test/blob/feature/az/bdp/tests/bdp_util.erl#L77.

I suggest you just use `bdp_util:build_cluster/1`, which incorporates all that's needed for a cluster to become ready.

File: tests/bdp_spark_test.erl [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    timer:sleep(20000),
+   
+    %% add and start master service to node1
+    ok = bdp_util:service_added(Node1, ?SPARK_MASTER_SERVICE_NAME, ?SPARK_MASTER_SERVICE_TYPE, ?SPARK_MASTER_SERVICE_CONFIG),
+    lager:info("Service ~p (~s) added", [?SPARK_MASTER_SERVICE_NAME, ?SPARK_MASTER_SERVICE_TYPE]),
+    lager:info("Waiting 10 seconds..."),
```
Comment (hmmr) [inline tests/bdp_spark_test.erl]: Similar to the above, there is `bdp_util:wait_services/2`, specifically to ensure the cluster has the requested services running: https://github.com/basho/riak_test/blob/feature/az/bdp/tests/bdp_util.erl#L94

File: tests/bdp_spark_test.erl [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    timer:sleep(10000),
+
+    ok = bdp_util:service_started(Node1, Node1, ?SPARK_MASTER_SERVICE_NAME, ?SPARK_MASTER_SERVICE_TYPE),
```
Comment (hmmr) [inline tests/bdp_spark_test.erl]: Lastly, some time ago I have renamed these `service_started` into `start_service`, for clarity.

My bad, I did this in another branch (https://github.com/basho/riak_test/tree/feature/vk/spark-master-test), but forgot to merge the changes back into mine.

You will have to pull from feature/az/bdp, do the appropriate changes in your commit, probably rebase your branch on top of feature/az/bdp, and push again (with --force).

File: tests/bdp_spark_test.erl [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    %%run spark job pi.py
+    SparkJobSubmit1 = "./lib/data_platform-1/priv/spark-master/bin/spark-submit --master ",
+    SparkJobSubmit2 = " ./lib/data_platform-1/priv/spark-master/examples/src/main/python/pi.py 100",
+    SparkJobSubmit3 = SparkJobSubmit1++SparkMasterAddr++SparkJobSubmit2,
+    Results = rpc:call(Node2,os,cmd,[SparkJobSubmit3]),
+    lager:info("Spark Job Results: ~s", [Results]),
+
+    %Test if the spark job submission worked
+    ?assert(string:str(Results,"Pi is roughly") > 0),
```
Comment (hmmr) [inline tests/bdp_spark_test.erl]: There's no such function: `string:str`. If you want a string to be converted into an integer, use `list_to_integer/1`.

File: tests/bdp_spark_test.erl (hunk 1 above)
Comment (korry8911) [inline tests/bdp_spark_test.erl]: i will replace sleeps with wait_for functions in the new PR

File: tests/bdp_spark_test.erl (hunk 4 above)
Comment (korry8911) [inline tests/bdp_spark_test.erl]: the purpose of using string:str is to check for successful completion of the spark job by looking for substring in successfull output.  Here is the docs for string:str http://www.erlang.org/doc/man/string.html

Comment (hmmr) [inline tests/bdp_spark_test.erl]: Ah, again my bad. Your code is correct here.

**Pull request closed without merging**